doc/develop/api/api_lifecycle.rst

```diff
@@ -236,9 +236,8 @@ The following are the requirements for deprecating an existing API:
   - Code using the deprecated API needs to be modified to remove usage of said
     API
   - The change needs to be atomic and bisectable
-  - Create a GitHub issue to track the removal of the deprecated API, and
-    add it to the roadmap targeting the appropriate release
-    (in the example above, 1.16).
+  - Add an entry in the `GitHub issue <https://github.com/zephyrproject-rtos/zephyr/issues/76943>`_
+    tracking removal of deprecated APIs.
```

Comment:
I think we should have one GitHub issue per release, documenting which deprecated functionality is to be removed prior to that release instead of having a singleton issue, which needs editing all the time.

The benefit of having one GitHub issue per release for tracking this, is that each issue can be added to the given milestone and easy tracked during release meetings, release checklists and so forth.

Reply:
Having an issue per release we cannot link at it from the docs or code like this PR is doing.
Also somebody needs to create that issue for each release, and that person is very likely not going to find a milestone ready.
So I'd expect what will happen if we do that is that the first few items which will be deprecated in each release will not be noted anywhere because there were too many steps.
Just like now nobody is creating an issue for each deprecation and setting it in the release plan to the right release.

Reply:
If you wanted, we could have developers add to this issue, and at the start of each release cut out the section for that release into a separate issue.
(Developers should normally only add 2 releases ahead)

Reply:
Correct me if I'm wrong here, I believe the proposed solution was to use the single issue to contain a list of the next few releases and the expected deprecation items for them.  At release time, one of the checklist items for the release managers is to copy the issue and increment the release version it tracks (i.e. drop 4.0 and add 4.3), and then bump this file version forward on the issue to track.  Does that sound correct?

Reply:
The PR has been changed to what was discussed and agreed in yesterday's release meeting.
@henrikbrixandersen @dkalowsk please re-review